src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,397 @@
+    /**
+     * {@inheritdoc}
+     */
+    public function write(TranslatorBagInterface $translatorBag): void
+    {
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            foreach ($catalogue->getDomains() as $domain) {
+                $content = $this->xliffFileDumper->formatCatalogue($catalogue, $domain, ['default_locale' => $this->defaultLocale]);
+
+                $fileId = $this->getFileId($domain);
+
+                if ($catalogue->getLocale() === $this->defaultLocale) {
+                    if (!$fileId) {
+                        $this->addFile($domain, $content);
+                    } else {
+                        $this->updateFile($fileId, $domain, $content);
+                    }
+                } else {
+                    $this->uploadTranslations($fileId, $domain, $content, $catalogue->getLocale());
+                }
+            }
+        }
+    }
+
+    public function read(array $domains, array $locales): TranslatorBag
+    {
+        $translatorBag = new TranslatorBag();
+
+        foreach ($domains as $domain) {
+            $fileId = $this->getFileId($domain);
+
+            if (!$fileId) {
+                continue;
+            }
+
+            foreach ($locales as $locale) {
+                if ($locale !== $this->defaultLocale) {
+                    $content = $this->exportProjectTranslations($locale, $fileId);
+                } else {
+                    $content = $this->downloadSourceFile($fileId);
+                }
+
+                $translatorBag->addCatalogue($this->loader->load($content, $locale, $domain));
+            }
+        }
+
+        return $translatorBag;
+    }
+
+    public function delete(TranslatorBagInterface $translatorBag): void
+    {
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            if ($catalogue->getLocale() !== $this->defaultLocale) {
+                continue;
+            }
+
+            foreach ($catalogue->getDomains() as $domain) {
+                $fileId = $this->getFileId($domain);
+
+                if (!$fileId) {
+                    continue;
+                }
+
+                $stringsMap = $this->mapStrings($fileId);
+
+                foreach ($catalogue->all() as $messages) {
+                    foreach ($messages as $id => $message) {
+                        if (!isset($stringsMap[$id])) {
+                            continue;
+                        }
+
+                        $this->deleteString($stringsMap[$id]);
+                    }
+                }
+            }
+        }
+    }
+
+    private function getFileId(string $domain): int
+    {
+        if (isset($this->files[$domain]) && $this->files[$domain]) {
+            return $this->files[$domain];
+        }
+
+        $files = $this->getFilesList();
+
+        foreach ($files as $file) {
+            if ($file['data']['name'] === sprintf('%s.%s', $domain, 'xlf')) {
+                $this->files[$domain] = (int) $file['data']['id'];
```

Comment:
you should avoid the explicit type cast here. If Crowdin changes their API to return non-integer ids, we should rather get a TypeError triggered by the return type than silently messing it due to the explicit cast (which will happily cast non-numeric strings)